src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/TradesMigrationService.cs

```diff
@@ -0,0 +1,185 @@
+                                    smallerCandles = currentCandles;
+
+                                // Preffered to perform inserting in parralel style.
+                                batchInsertTasks.Add(
+                                    _candlesHistoryRepository.InsertOrMergeAsync(
+                                        currentCandles.Candles.Values,
+                                        currentCandles.AssetId,
+                                        TradesCandleBatch.PriceType,
+                                        currentCandles.TimeInterval));
+                            }
+
+                            _tradesMigrationHealthService[searchToken.AssetPairId].SummaryFetchedTrades += batchCount;
+
+                            await Task.WhenAll(batchInsertTasks.ToArray());
+
+                            await _log.WriteInfoAsync(nameof(TradesMigrationManager), nameof(MigrateTradesCandlesAsync),
+                                $"Batch of {batchCount} records for {searchToken.AssetPairId} processed.");
+                        }
+
+                        await _log.WriteInfoAsync(nameof(TradesMigrationManager), nameof(MigrateTradesCandlesAsync),
+                            $"Migration for {searchToken.AssetPairId} finished. Total records migrated: {_tradesMigrationHealthService[searchToken.AssetPairId].SummaryFetchedTrades}, " +
+                            $"total candles stored: {_tradesMigrationHealthService[searchToken.AssetPairId].SummarySavedCandles}.");
+                    }
+                    catch (Exception ex)
+                    {
+                        _tradesMigrationHealthService.State = TradesMigrationState.Error;
+                        await _log.WriteErrorAsync(nameof(TradesMigrationManager), nameof(MigrateTradesCandlesAsync), ex);
+                        await _log.WriteInfoAsync(nameof(TradesMigrationManager), nameof(MigrateTradesCandlesAsync),
+                            $"Migration for {searchToken.AssetPairId} interrupted due to error. Please, check error log for details.");
+                        return;
+                    }
+                }
+            }
+
+            _tradesMigrationHealthService.State = TradesMigrationState.Finished;
+        }
+
+        #endregion
+
+        #region Private
+
+        /// <summary>
+        /// Asynchronous trades candles removal from all the time interval tables for the specified asset pair.
+        /// </summary>
+        private async Task RemoveTradesCandlesAsync(string assetPairId, DateTime? removeByDate)
+        {
+            var removalTasks = new List<Task>();
+
+            foreach (var interval in Candles.Constants.StoredIntervals)
+                removalTasks.Add(_candlesHistoryRepository.DeleteCandlesAsync(assetPairId, interval, CandlePriceType.Trades, null, removeByDate)); // Remove since the oldest candles till removeByDate.
+
+            await Task.WhenAll(removalTasks);
```

Comment:
just return task without await

Reply:
If I remove await from here, the method will be executed synchronously (for this is the single await in it). From the other side, it does not seem much comfortable to return a list of tasks to the caller code. I'm confused. Let it remain unchanged for a while...